### PR TITLE
1578 commit to bb

### LINKED ISSUE #1578: Commit to BB
When commiting changes from simulation, show a dialog:

- "Are you sure"`?
- State the name of the PV BB and the module where the values will be written into.

### PR DESCRIPTION
**Pull request by benjaperez1983**
Fixes #1578

# Description

Add message with confirmation and information when commiting to BB From simulation

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):
If last module has building blocks selected
image

if no building blocks are selected
image

# Questions (if appropriate):

### CODE REVIEW

File: src/MoBi.Presentation/Tasks/SimulationCommitTask.cs [hunk 1]
```diff
@@ -91,11 +109,11 @@ public IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChange
       ///    have SelectedParameterValues as that building block will receive the updates. The template module and building block
       ///    is resolved from the project by name
       /// </summary>
-      private IEnumerable<IMoBiCommand> updateParameterValuesFromSimulationChanges(IMoBiSimulation simulation, ModuleConfiguration moduleConfiguration)
+      private IEnumerable<IMoBiCommand> updateParameterValuesFromSimulationChanges(ModuleConfiguration moduleConfiguration, IReadOnlyList<(ObjectPath quantityPath, Parameter quantity)> parameterChanges)
       {
          var templateBuildingBlock = _templateResolverTask.TemplateBuildingBlockFor(moduleConfiguration.SelectedParameterValues);
 
-         var valueTuples = changesFrom<Parameter>(simulation).ToList();
+         var valueTuples = parameterChanges;
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Tasks/SimulationCommitTask.cs:116]: you don't need that assignment now

File: src/MoBi.Presentation/Tasks/SimulationCommitTask.cs [hunk 2]
```diff
@@ -107,11 +125,11 @@ private IEnumerable<IMoBiCommand> updateParameterValuesFromSimulationChanges(IMo
       ///    have SelectedInitialConditions as that building block will receive the updates. The template module and building
       ///    block is resolved from the project by name
       /// </summary>
-      private IEnumerable<IMoBiCommand> updateInitialConditionsFromSimulationChanges(IMoBiSimulation simulation, ModuleConfiguration moduleConfiguration)
+      private IEnumerable<IMoBiCommand> updateInitialConditionsFromSimulationChanges(ModuleConfiguration moduleConfiguration, IReadOnlyList<(ObjectPath quantityPath, MoleculeAmount quantity)> moleculeChanges)
       {
          var templateBuildingBlock = _templateResolverTask.TemplateBuildingBlockFor(moduleConfiguration.SelectedInitialConditions);
 
-         var valueTuples = changesFrom<MoleculeAmount>(simulation).ToList();
+         var valueTuples = moleculeChanges;
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/Tasks/SimulationCommitTask.cs:132]: or that one.

File: src/MoBi.Presentation/UICommand/CommitSimulationUICommand.cs [hunk 3]
```diff
@@ -18,7 +18,9 @@ public CommitSimulationUICommand(IMoBiContext context, IActiveSubjectRetriever a
 
       protected override void PerformExecute()
       {
-         _context.AddToHistory(_simulationCommitTask.CommitSimulationChanges(Subject));
+         var command = _simulationCommitTask.CommitSimulationChanges(Subject);
+         if(command!= null)
```
Comment (rwmcintosh) [inline src/MoBi.Presentation/UICommand/CommitSimulationUICommand.cs:22]: just fix up the spacing on `command!= null`

**Review by rwmcintosh — APPROVED** [review]
Just fix up those minor comments and add Juri.

**Review by rwmcintosh — CHANGES_REQUESTED** [review]
I didn't realize that the build has failed

**Comment by rwmcintosh** [comment]
> I didn't realize that the build has failed

It's probably because you need a fake on the dialog box

**Review by rwmcintosh — APPROVED** [review]

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**